### PR TITLE
Validate aps-environment in entitlements

### PR DESCRIPTION
**Pull request by keith**

### CODE REVIEW

**Comment by thomasvl** [comment]
Actually, can you add tests for this in the unittest, this is one tool we've got pretty good coverage on.

**Comment by keith** [comment]
@thomasvl added some tests!

File: tools/plisttool/plisttool.py [hunk 1]
```diff
@@ -1166,7 +1166,7 @@ def _validate_entitlements_against_profile(
             **report_extras)
 
     aps_environment = entitlements.get('aps-environment')
-    if aps_environment and profile_entitlements:
+    if aps_environment and profile_entitlements is not None:
```
[the remark this answers is not in the document]
Comment (keith) [reply tools/plisttool/plisttool.py]: So the problem was this test case: `'Entitlements': {},`. In this case this wouldn't hit, and would be a false negative. I'm not sure if having a literally empty profile entitlements section is possible, maybe you at least have to have a bundle ID? If so I can add that instead and remove this.

[the remark this answers is not in the document]
Comment (thomasvl) [reply tools/plisttool/plisttool.py]: Hopefully Apple wouldn't create one like that.  😦 

The other option is just be consistent in all our code to always do the not None check.  It just sorta sticks out which is a little confusing.

[the remark this answers is not in the document]
Comment (keith) [reply tools/plisttool/plisttool.py]: For now I've updated the test to have a different key instead. If we hit this case in the future we can add this to all of them

**Comment by keith** [comment]
rebased!

**Comment by thomasvl** [comment]
Thanks, testing this internally also.  The PR will close when it lands and is pushed back out.

**Pull request closed without merging**